cf-custom-resources/lib/custom-domain.js

```diff
@@ -0,0 +1,355 @@
+  var responseData = {};
+  var physicalResourceId;
+  const props = event.ResourceProperties;
+  const [app, env, domain] = [props.AppName, props.EnvName, props.DomainName];
+  var aliasTypes = {
+    EnvDomainZone: {
+      regex: `.*${env}.${app}.${domain}`,
+      domain: `${env}.${app}.${domain}`,
+    },
+    AppDomainZone: { regex: `.*${app}.${domain}`, domain: `${app}.${domain}` },
+    RootDomainZone: { regex: `.*${domain}`, domain: `${domain}` },
+    OtherDomainZone: { regex: `.*` },
+  };
+  try {
+    var aliases = await getAllAliases(props.Aliases);
+    switch (event.RequestType) {
+      case "Create":
+        await writeCustomDomainRecord(
+          aliases,
+          props.LoadBalancerDNS,
+          props.LoadBalancerHostedZone,
+          props.AppDNSRole,
+          aliasTypes,
+          "UPSERT"
+        );
+        break;
+      case "Update":
+        await writeCustomDomainRecord(
+          aliases,
+          props.LoadBalancerDNS,
+          props.LoadBalancerHostedZone,
+          props.AppDNSRole,
+          aliasTypes,
+          "UPSERT"
+        );
+        // After upserting new aliases, delete unused ones. For example: previously we have ["foo.com", "bar.com"],
+        // and now the aliases param is updated to just ["foo.com"] then we'll delete "bar.com".
+        var prevAliases = await getAllAliases(event.OldResourceProperties.Aliases);
+        var aliasesToDelete = prevAliases.filter(function (itm) {
+          return aliases.indexOf(itm) === -1;
+        });
+        await writeCustomDomainRecord(
+          aliasesToDelete,
+          props.LoadBalancerDNS,
+          props.LoadBalancerHostedZone,
+          props.AppDNSRole,
+          aliasTypes,
+          "DELETE"
+        );
+        break;
+      case "Delete":
+        await writeCustomDomainRecord(
+          aliases,
+          props.LoadBalancerDNS,
+          props.LoadBalancerHostedZone,
+          props.AppDNSRole,
+          aliasTypes,
+          "DELETE"
+        );
+        break;
+      default:
+        throw new Error(`Unsupported request type ${event.RequestType}`);
+    }
+    physicalResourceId = `${event.LogicalResourceId}`;
+    await report(event, context, "SUCCESS", physicalResourceId, responseData);
+  } catch (err) {
+    console.log(`Caught error ${err}.`);
+    await report(
+      event,
+      context,
+      "FAILED",
+      physicalResourceId,
+      null,
+      err.message
+    );
+  }
+};
+
+// getAllAliases gets all aliases out from a string. For example:
+// {"frontend": ["test.foobar.com", "foobar.com"], "api": ["api.foobar.com"]} will return
+// ["foobar.com"].
+const getAllAliases = function (aliases) {
+  let obj;
+  try {
+    obj = JSON.parse(aliases || "{}");
+  } catch (error) {
+    throw new Error(`Cannot parse ${aliases} into JSON format.`);
+  }
+  var aliasList = [];
+  for (var m in obj) {
+    aliasList.push(...obj[m]);
+  }
+  return [...new Set(aliasList)];
```

Comment:
nit: can we just return a Set object instead of transforming into a list again? this way the following operation above can be faster 🏃 
```js
var aliasesToDelete = prevAliases.filter(function (itm) {
     return aliases.indexOf(itm) === -1;
});
```